scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -250,14 +255,15 @@ public boolean isSuccess() {
   }
 
   /**
-   * Supplier for {@link AsyncLookupDoFn} with caching logic.
+   * Supplier for {@link BaseAsyncLookupDoFn} with caching logic.
```

Comment:
Isn't this technically supplying a `Cache` not a `BaseAsyncLookupDoFn` ? 🤔 Also -- it seems a bit strange to me that this class is both a Supplier for new `Cache`s while also handling get/put logic for a cache of `Cache`s? It seems like `CacheManager` might be a more accurate name... it's not a big deal though.

Reply:
Maybe rephrase it as `{@link Cache} supplier for {@link BaseAsyncLookupDoFn}`? Supplier is from the Java naming convention for `() => T`.

Reply:
oh, yeah my thought here was more that the scope of this class seems to be greater than just being a `Supplier`! I think your rephrasing makes sense though.